src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -17,10 +17,227 @@ namespace System
+        /// </summary>
+        /// <param name="source">The source span from which the characters are removed.</param>
+        /// <param name="startIndex">The zero-based position to begin deleting characters.</param>
+        /// <param name="count">The number of characters to delete.</param>
+        /// <param name="destination">The span to copy the leftover characters into.</param>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when the destination Span is too small to contain the source Span with the characters removed
+        /// (i.e. destination.Length &lt; source.Length - count).
+        /// </exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when either <paramref name="startIndex"/> or <paramref name="count"/> is less than zero or
+        /// when <paramref name="startIndex"/> plus <paramref name="count"/> specify a position outside the source span.
+        /// </exception>
+        public static void Remove(this ReadOnlySpan<char> source, int startIndex, int count, Span<char> destination)
+        {
+            if ((uint)startIndex > (uint)source.Length || (uint)count > (uint)(source.Length - startIndex))
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+            if (destination.Length < source.Length - count)
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
+
+            // Nothing to remove
+            if (count == 0)
+            {
+                source.CopyTo(destination);
+                return;
+            }
+
+            if (source.Length - count != 0)
+            {
+                if (startIndex == 0)
+                {
+                    source.Slice(count).CopyTo(destination);
+                    return;
+                }
+
+                // destination is after source, reverse the order of copying
+                if (source.Overlaps(destination, out int elementOffset) && elementOffset > 0)
+                {
+                    if (startIndex + count != source.Length)
+                        source.Slice(startIndex + count).CopyTo(destination.Slice(startIndex));
+
+                    // The Slice is necessary since startIndex cannot equal source.Length here.
+                    source.Slice(0, startIndex).CopyTo(destination);
+                }
+                else
+                {   // no overlap or destination is before source
+                    // The Slice is necessary since startIndex cannot equal source.Length here.
+                    source.Slice(0, startIndex).CopyTo(destination);
+
+                    if (startIndex + count != source.Length)
+                        source.Slice(startIndex + count).CopyTo(destination.Slice(startIndex));
+                }
+            }
+            else
+            {   // Remove everything
```

Comment:
Do we really want to pay the cost of clearing here. Why?

Reply:
We are only clearing whatever falls within source.Length - count. Nothing beyond that is being cleared. See tests like https://github.com/dotnet/corefx/pull/26598/files#diff-e1048de19afdbf1765ab34e0dfea880cR255